lib/global_id/locator.rb

```diff
@@ -7,14 +7,23 @@ class << self
       # Takes either a GlobalID or a string that can be turned into a GlobalID
       #
       # Options:
+      # * <tt>:includes</tt> - A Symbol, Array, Hash or combination of them
+      #   The same structure you would pass into a includes method of ActiveRecord.
+      #   @see https://guides.rubyonrails.org/active_record_querying.html#eager-loading-associations
+      #   If present, locate will load all the relationships specified here.
       # * <tt>:only</tt> - A class, module or Array of classes and/or modules that are
       #   allowed to be located.  Passing one or more classes limits instances of returned
       #   classes to those classes or their subclasses.  Passing one or more modules in limits
       #   instances of returned classes to those including that module.  If no classes or
       #   modules match, +nil+ is returned.
       def locate(gid, options = {})
-        if gid = GlobalID.parse(gid)
-          locator_for(gid).locate gid if find_allowed?(gid.model_class, options[:only])
+        gid = GlobalID.parse(gid)
+        return unless gid && find_allowed?(gid.model_class, options[:only])
+
+        if locator_for(gid).method(:locate).arity == 1
+          locator_for(gid).locate(gid)
```

Comment:
We should deprecate this branch and tell people to always receive two arguments in the locator.

Reply:
@rafaelfranca, just asking; what about using `keyword arguments` instead of forcing the two arguments always being passed?